openlibrary/plugins/openlibrary/js/check-ins/index.js

```diff
@@ -591,19 +593,24 @@ function fetchProgressAndUpdateView(yearlyGoalElem, goalYear) {
         })
         .then(function(data) {
             const html = data['partials']
-            const progress = document.createElement('SPAN')
-            progress.id = 'reading-goal-container'
-            progress.innerHTML = html
-            yearlyGoalElem.appendChild(progress)
-
-            // Hide the "Set 20XX reading goal" link:
-            yearlyGoalElem.children[0].classList.add('hidden')
-
-            const progressEditLink = progress.querySelector('.edit-reading-goal-link')
-            const updateModal = progress.querySelector('dialog')
-            initDialogs([updateModal])
-            addGoalEditClickListener(progressEditLink, updateModal)
-            const submitButton = updateModal.querySelector('.reading-goal-submit-button')
-            addGoalSubmissionListener(submitButton)
+            yearlyGoalElems.forEach((yearlyGoalElem) => {
+                const progress = document.createElement('SPAN')
+                progress.id = 'reading-goal-container'
+                progress.innerHTML = html
+                yearlyGoalElem.appendChild(progress)
+
+                // Hide the "Set 20XX reading goal" link:
+                const link = yearlyGoalElem.querySelector('.set-reading-goal-link');
```

Comment:
```suggestion
                // Hide the desktop "Set 20XX reading goal" link:
                const link = yearlyGoalElem.querySelector('.set-reading-goal-link:not(.li-title-desktop)');
```
Small update to avoid selecting the mobile `.set-reading-goal-link`, which also serves as a section header.